src/neuroconv/datainterfaces/ecephys/spikeglx/spikeglxnidqinterface.py

```diff
@@ -72,18 +74,36 @@ def __init__(
         if folder_path is not None:
             self.folder_path = Path(folder_path)
 
+        from spikeinterface.extractors import SpikeGLXRecordingExtractor
+
+        self.recording_extractor = SpikeGLXRecordingExtractor(
+            folder_path=self.folder_path,
+            stream_id="nidq",
+            all_annotations=True,
+        )
+
+        channel_ids = self.recording_extractor.get_channel_ids()
+        analog_channel_signatures = ["XA", "MA"]
+        self.analog_channel_ids = [ch for ch in channel_ids if "XA" in ch or "MA" in ch]
+        self.has_analog_channels = len(self.analog_channel_ids) > 0
+        self.has_digital_channels = len(self.analog_channel_ids) < len(channel_ids)
+        if self.has_digital_channels:
+            import ndx_events  # noqa: F401
+            from spikeinterface.extractors import SpikeGLXEventExtractor
+
+            self.event_extractor = SpikeGLXEventExtractor(folder_path=self.folder_path)
+
         super().__init__(
             verbose=verbose,
             load_sync_channel=load_sync_channel,
             es_key=es_key,
```

Comment:
should es_key still be here? "es" stands for ElectricalSeries

Reply:
Yes, I think you are right. 

At the moment, it is not clear how to pass metadata to this object but also there is not clear metadata to be passed. I discussed yesterday with a user that has this type of data about what could be useful for their use case and the only thing that came up is a description of the channels. 

I will open another issue to discuss this.

Related, I really like the proposal that you had here:
https://github.com/catalystneuro/neuroconv/issues/629

Reply:
OK, I opened an issue here:
https://github.com/catalystneuro/neuroconv/issues/1154

I would like to hear your thoughts on it when you have some time.